core/dbt/cli/main.py

```diff
@@ -349,6 +350,7 @@ def run(ctx, **kwargs):
 
 # dbt run operation
 @cli.command("run-operation")
+@click.argument("macro")
```

Comment:
tiny nit: could we move this down under `@click.pass_context` so that all the params are logically grouped together?

Reply:
also - we should probably define this in `core/dbt/cli/params.py` alongside the other params for consistency / reuse

Reply:
Agree with first nit! I'll make the order match the CLI order

For the second - two reasons I prefer keeping it alongside the command definitions:
1. Click arguments can't have documentation / help text, [click's recommendation](https://click.palletsprojects.com/en/8.1.x/documentation/#documenting-arguments) is to document them by mentioning them explicitly in the command help text
2. We have very very few positional arguments in dbt commands. (I think it's just `run-operation MACRO` and `init PROJECT_NAME`. I just remembered the latter, and will add in this PR so we don't forget.) These are truly one-offs. IMO adding them as (reusable) params risks the misperception that they're widely used, versus keeping them alongside the only commands for which they're relevant.